### ISSUE TITLE
E2: io* functions have to be fixed

### ISSUE THREAD

**Issue by CaptainPRICE** [wiremod/wire#1420]
```golo
@name A
@inputs A
@outputs A
@persist A
@trigger A
A = 1
local A = 2
print(A)
A = 3
if (A) {
    local A = 4
    function normal a() { return A }
    function normal a(A) { return A * 2 }
    function normal a(A, B) { local A = A * ioSetOutput(entity():getName(), B * random()),return A + B * 4 }
    print(A, a(), a(a(A), A * curtime()))
}
print(A, "-A: " +-ioSetOutput(entity():getName(), 1337))
```

If you run above E2, you will receive the following runtime error:
> sv: Expression 2 (A): entities/gmod_wire_expression2/core/number.lua:157: attempt to perform arithmetic on a nil value

That is because `ioSetOutput` function says that it returns a value, but it actually does not, thus it throws a nil value error (first error is thrown at line 14, E2 code). So, I fixed the `a(nn)` UDF:
`function normal a(A, B) { local C = B * random(),ioSetOutput(entity():getName(), C),local A = A * C,return A + B * 4 }`
But, then yet another runtime error has been thrown:
> sv: Expression 2 (A): entities/gmod_wire_expression2/core/number.lua:140: attempt to perform arithmetic on a nil value

That one is because of the line 17 in the E2, so I have fixed that like so:
`local C = 1337,ioSetOutput(entity():getName(), C),print(A, "-A: " +-C)`

And guess what, it finally works. Here is the final E2:
```golo
@name A
@inputs A
@outputs A
@persist A
@trigger A
A = 1
local A = 2
print(A)
A = 3
if (A) {
    local A = 4
    function normal a() { return A }
    function normal a(A) { return A * 2 }
    function normal a(A, B) { local C = B * random(),ioSetOutput(entity():getName(), C),local A = A * C,return A + B * 4 }
    print(A, a(), a(a(A), A * curtime()))
}
local C = 1337,ioSetOutput(entity():getName(), C),print(A, "-A: " +-C)
```

-----

This is related to #1374. I will see if I can fix it in #1375, later.

**Comment by Divran** [comment]
You already posted a pr where you fixed all functions that return the wrong
value. However, that pr was rejected because you also added lots of
pointless syntax changes. I suggest you update that pr to ONLY fix bugs and
leave out all syntax changes. Then we can merge it.

**Comment by adosikas** [comment]
Request to close this as invalid, this is a duplicate of #1375 which also includes the issue I now described in #1421.

**Comment by CaptainPRICE** [comment]
Just ignore it?
The issue will be automatically closed once the PR is merged (but I have to fix things up there still, as @Divran has pointed out). I have created this issue to remind me later, and to let everyone else know about it (in a separate issue).
So, I am against closing this or even marking it as invalid. "Duplicate" is fine tho.
